src/services/createIsSameHost.ts

```diff
@@ -0,0 +1,16 @@
+import { createMaybeRelativeUrl } from '@/services/createMaybeRelativeUrl'
+
+export function createIsSameHost(host: string | undefined): (url: string) => boolean {
```

Comment:
What do you think about this being "createIsExternal"? I think generally that's what we want to know and it might be nice to carry that logic through to routes and router-link.